CodeU_Assignement_2/test/codeu_assignement_2/KemBinaryTreeTest.java

```diff
@@ -0,0 +1,215 @@
+package codeu_assignement_2;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import static org.junit.Assert.*;
+
+/**
+ *
+ * @author Karine
+ */
+public class KemBinaryTreeTest {
+    
+    public KemBinaryTreeTest() {
+    }
+    
+    @BeforeClass
+    public static void setUpClass() {
+    }
+    
+    @AfterClass
+    public static void tearDownClass() {
+    }
+
+    /**
+     * Test of isEmpty method, of class KemBinaryTree.
+     */
+    @Test
+    public void testIsEmpty() {
+        System.out.println("isEmpty");
+        KemBinaryTree instance = new KemBinaryTree();
+        boolean expResult = true;
+        boolean result = instance.isEmpty();
+        assertEquals("New tree shall be empty", expResult, result);
+    }
+
+    /**
+     * Test of getRoot method, of class KemBinaryTree.
+     */
+    @Test
+    public void testGetRoot() {
+        System.out.println("getRoot");
+        KemBinaryTree instance = new KemBinaryTree();
+        KemTreeNode expResult = null;
+        KemTreeNode result = instance.getRoot();
+        assertEquals("Empty tree doesn't have null root.", expResult, result);
+    }
+
+    /**
+     * Test of printAncestors method, of class KemBinaryTree.
+     */
+    @Test
+    public void testPrintAncestorsNull() {
+        System.out.println("printAncestors");
+        KemTreeNode child = null;
+        KemBinaryTree instance = new KemBinaryTree();
+        instance.printAncestors(child);
+    }
+    @Test
+    public void testPrintAncestors() {
+        System.out.println("printAncestors");
+        
+        /* Create tree for testing */
+        KemBinaryTree<Integer> t1 = new KemBinaryTree();
+        Integer[] temp = {1,2,3,4,5,6,7,8,9,0,1,2,3,4,5};
+        ArrayList<Integer> arr1 = new ArrayList();
+        arr1.addAll(Arrays.asList(temp));
+        t1.addAll(arr1);
+        
+        t1.printAncestors(t1.getRoot().getLeft().getLeft().getLeft());
+        t1.printAncestors(t1.getRoot().getLeft().getLeft().getRight());
+        t1.printAncestors(t1.getRoot().getLeft().getRight().getRight());
+        t1.printAncestors(t1.getRoot().getRight().getRight().getRight());
+        t1.printAncestors(t1.getRoot());
+        t1.printAncestors(t1.getRoot().getRight().getRight().getRight().getRight());
+    }

```

Comment:
This doesn't actually check that the results are as expected. To improve testability, I would split the algorithm into a part that generates a string of the ancestors, and a wrapper that calls the method and prints the string. Using this, you can do proper unit testing on the method that generates the string.